client/components/email-verification/email-verification-dialog/index.jsx

```diff
@@ -0,0 +1,118 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import noop from 'lodash/noop';
+import { localize } from 'i18n-calypso';
+import { connect } from 'react-redux';
+import { get, includes } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import Dialog from 'components/dialog';
+import Button from 'components/button';
+import Spinner from 'components/spinner';
+import { getCurrentUserEmail } from 'state/current-user/selectors';
+import {
+	verifyEmail,
+	resetVerifyEmailState,
+} from 'state/current-user/email-verification/actions';
+
+class VerifyEmailDialog extends Component {
+
+	getResendButtonLabel() {
+		if ( 'sent' === this.props.emailVerification ||
+			'error' === this.props.emailVerification ) {
+			return this.props.translate( 'Email Sent' );
+		}
+		if ( 'requesting' === this.props.emailVerification ) {
+			return <Spinner className="email-verification-dialog__confirmation-dialog-spinner" />;
+		}
+		return this.props.translate( 'Resend Email' );
+	}
```

Comment:
Probably worth pushing to a subsequent PR? The changes in this PR are now getting pretty long, and it's blocking a bunch of other PRs.